src/blenderbim/blenderbim/bim/module/aggregate/operator.py

```diff
@@ -190,12 +190,17 @@ class BIM_OT_select_parts(bpy.types.Operator):
 
     def execute(self, context):
         self.file = IfcStore.get_file()
-        obj = bpy.data.objects.get(self.obj) or context.active_object
-        parts = ifcopenshell.util.element.get_parts(tool.Ifc.get_entity(obj))
-        parts_objs = set(tool.Ifc.get_object(part) for part in parts)
-        selectable_parts_objs = set(context.selectable_objects).intersection(parts_objs)
-        for selectable_part_obj in selectable_parts_objs:
-            selectable_part_obj.select_set(True)
+        # obj = bpy.data.objects.get(self.obj) or context.active_object
+
+        for obj in context.selected_objects:
+            parts = ifcopenshell.util.element.get_parts(tool.Ifc.get_entity(obj))
+            if parts:
+                p_objs = set(tool.Ifc.get_object(part) for part in parts)
+                selectable_p_objs = set(context.selectable_objects).intersection(p_objs)
```

Comment:
My only comment would be to keep the name of the variables for clarity. `parts_objs` and `selectable_parts_objs`.